### ISSUE TITLE
response-headers endpoint adding second Content-Type header rather than replacing

### ISSUE THREAD

**Issue by tmenier** [kennethreitz/httpbin#347]
Any call to `response-headers` appears to be returning `Content-Type: application/json` (in both the actual response headers and the JSON body). If you specify a `Content-Type` in the query string, it will add a second header of the same name rather than replacing the existing one. This is confirmed by the example on your site:

https://httpbin.org/response-headers?Content-Type=text%2Fplain%3B+charset%3DUTF-8&Server=httpbin

Is this behavior intentional? I believe this changed sometime in about the past month. In .NET, `HttpResponseMessage.Content.Headers.ContentType` will grab the first `Content-Type` header if there are multiple, and I'm using this in some tests that hit httpbin and had been passing until very recently. I dug through the history of core.py and unfortunately wasn't able to pin it on a specific commit (my Python chops are not sharp).

Thanks!

**Comment by sigmavirus24** [comment]
@tmenier here's a hard question: What does httpbin return? When you give it a random content-type? Should it always return a JSON body but return the content-type specified by the user? That seems like a lie. Does it try to return the content-type specified? If so how do we format things for plain-text, jpeg, or any other binary format? I think the behaviour here is not intuitive but I'm also not sure what the behaviour should be in this case.

**Comment by tmenier** [comment]
In my opinion, trying to conform the response body to the header I want returned is beyond the scope of what that endpoint should do. I think `/response-headers` ought to just blindly echo the headers I specify and not try to infer any meaning out of them. Yes you could say httpbin is lying in this case, but that's because I'm telling it to. :) I might specifically have a test that says "what happens when the API I'm calling lies?" I might also want to test what happens if a totally invalid content-type header is returned; I don't expect httpbin to care about that either.

Again, just my opinion.

**Comment by tmenier** [comment]
If it helps, [here's the test](https://github.com/tmenier/Flurl/blob/master/Test/Flurl.Test.Shared/Http/RealHttpTests.cs#L173-L188) in my project that recently started failing due to this issue.

**Comment by mortenlj** [comment]
I agree with @tmenier here, I have the exact same issue. I have code that should do different things depending on the `Content-Type` in the response. In my tests, I mock out the actual handling of the data, and test that the correct handler is called for the requested `Content-Type`. Now, the `Content-Type` is always interpreted as `application/json`, so all my tests are failing.
